### PR TITLE
[SCB-1127] Support alpha server registering to Spring Cloud Eureka

### PR DESCRIPTION
**Pull request by coolbeevip**
### Build Alpha Server with Spring Cloud Eureka
spring-boot 1.x
> spring-cloud-netflix-eureka-client-1.4.6.RELEASE.jar and dependencies will be added to alpha-server-0.4.0-SNAPSHOT-exec.jar
```shell
mvn clean package -DskipTests=true -Pspring-cloud-eureka,spring-boot-1
```
spring-boot 2.x
> spring-cloud-netflix-eureka-client-2.0.2.RELEASE.jar  and dependencies will be added to alpha-server-0.4.0-SNAPSHOT-exec.jar
```shell
mvn clean package -DskipTests=true -Pspring-cloud-eureka,spring-boot-2
```
### Registering with Eureka
Configuring startup parameters `spring.profiles.active=spring-cloud-eureka` and `eureka.client.service-url.defaultZone`, as shown in the following example

```shell
java -jar alpha-server-0.4.0-SNAPSHOT-exec.jar \
  --spring.datasource.platform=mysql \
  --spring.datasource.dataSourceClassName=com.mysql.jdbc.Driver \
  --spring.datasource.url="jdbc:mysql://127.0.0.1:3306/saga?useUnicode=true&characterEncoding=utf-8&autoReconnect=TRUE" \
  --spring.datasource.username=saga-user \
  --spring.datasource.password=saga-password \
  --spring.profiles.active=spring-cloud-eureka \
  --eureka.client.service-url.defaultZone=http://127.0.0.1:8761/eureka
```
Key information in the log file
```
[main] INFO  org.springframework.cloud.netflix.eureka.serviceregistry.EurekaServiceRegistry - Registering application SERVICECOMB-ALPHA-SERVER with eureka with status UP
```
### Get Alpha Server info with Eureka
```
curl http://127.0.0.1:8761/eureka/apps/
```
output
> See \<servicecomb-alpha-server\>0.0.0.0:8080\</servicecomb-alpha-server\> in the metadata fragment to indicate successful registration
```
<applications>
  <versions__delta>1</versions__delta>
  <apps__hashcode>UP_1_</apps__hashcode>
  <application>
    <name>SERVICECOMB-ALPHA-SERVER</name>
    <instance>
      <instanceId>10.50.7.15::servicecomb-alpha-server:8090</instanceId>
      <hostName>0.0.0.0</hostName>
      <app>SERVICECOMB-ALPHA-SERVER</app>
      <ipAddr>0.0.0.0</ipAddr>
      <status>UP</status>
      <overriddenstatus>UNKNOWN</overriddenstatus>
      <port enabled="true">8090</port>
      <securePort enabled="false">443</securePort>
      <countryId>1</countryId>
      <dataCenterInfo class="com.netflix.appinfo.InstanceInfo$DefaultDataCenterInfo">
        <name>MyOwn</name>
      </dataCenterInfo>
      <leaseInfo>
        <renewalIntervalInSecs>30</renewalIntervalInSecs>
        <durationInSecs>90</durationInSecs>
        <registrationTimestamp>1548214185871</registrationTimestamp>
        <lastRenewalTimestamp>1548214185871</lastRenewalTimestamp>
        <evictionTimestamp>0</evictionTimestamp>
        <serviceUpTimestamp>1548214185871</serviceUpTimestamp>
      </leaseInfo>
      <metadata>
        <management.port>8090</management.port>
        <servicecomb-alpha-server>0.0.0.0:8080</servicecomb-alpha-server>
      </metadata>
      <homePageUrl>http://0.0.0.0:8090</homePageUrl>
      <statusPageUrl>http://0.0.0.0:8090/info</statusPageUrl>
      <healthCheckUrl>http://0.0.0.0:8090/health</healthCheckUrl>
      <vipAddress>servicecomb-alpha-server</vipAddress>
      <secureVipAddress>servicecomb-alpha-server</secureVipAddress>
      <isCoordinatingDiscoveryServer>false</isCoordinatingDiscoveryServer>
      <lastUpdatedTimestamp>1548214185871</lastUpdatedTimestamp>
      <lastDirtyTimestamp>1548214185716</lastDirtyTimestamp>
      <actionType>ADDED</actionType>
    </instance>
  </application>
</applications>
```

### CODE REVIEW

File: alpha/alpha-server/src/main/resources/application.yaml [hunk 1]
```diff
@@ -16,16 +16,19 @@
 ## ---------------------------------------------------------------------------
 server:
   port: 8090
+  host: 0.0.0.0
```
Comment (WillemJiang) [inline alpha/alpha-server/src/main/resources/application.yaml]: It's not a good address which could compromise our application.

File: alpha/alpha-server/src/main/resources/application.yaml [hunk 2]
```diff
@@ -65,3 +68,23 @@ spring:
...
+  instance:
+    metadataMap:
+      servicecomb-alpha-server: ${server.host}:${alpha.server.port}
+    ip-address: ${server.host}
+    preferIpAddress: true
+    nonSecurePortEnabled: true
+    securePortEnabled: false
+    homePageUrl: http://${server.host}:${server.port}
+    healthCheckUrl: http://${server.host}:${server.port}/health
+    statusPageUrl: http://${server.host}:${server.port}/info
```
Comment (WillemJiang) [inline alpha/alpha-server/src/main/resources/application.yaml]: It's spring-cloud 1.x address

File: alpha/alpha-server/pom.xml [hunk 3]
```diff
@@ -258,5 +258,15 @@
         </plugins>
       </build>
     </profile>
+    <profile>
+      <id>spring-cloud</id>
```
Comment (WillemJiang) [inline alpha/alpha-server/pom.xml]: My suggestion is using spring-cloud-consul by default. We can fix it later.

File: alpha/alpha-server/src/main/resources/application.yaml (hunk 2 above)
Comment (coolbeevip) [reply alpha/alpha-server/src/main/resources/application.yaml]: Non-required parameters are removed in application.yml

File: alpha/alpha-server/pom.xml (hunk 3 above)
Comment (coolbeevip) [reply alpha/alpha-server/pom.xml]: rename spring-cloud-eureka

**Pull request merged**